nvidia-gpu/nvidia-fabricmanager/lts/pkg.yaml

```diff
@@ -2,56 +2,93 @@ name: nvidia-fabricmanager-lts
         destination: fabricmanager.tar.xz
         sha256: {{ .NVIDIA_FABRIC_MANAGER_LTS_ARM64_SHA256 }}
         sha512: {{ .NVIDIA_FABRIC_MANAGER_LTS_ARM64_SHA512 }}
+      - url: https://developer.download.nvidia.com/compute/cuda/redist/nvlsm/linux-sbsa/nvlsm-linux-sbsa-{{ .NVIDIA_NVLSM_VERSION }}-archive.tar.xz
+        destination: nvlsm.tar.xz
+        sha256: {{ .NVIDIA_NVLSM_ARM64_SHA256 }}
+        sha512: {{ .NVIDIA_NVLSM_ARM64_SHA512 }}
     # {{ else }} This in fact is YAML comment, but Go templating instruction is evaluated by bldr
       - url: https://developer.download.nvidia.com/compute/nvidia-driver/redist/fabricmanager/linux-x86_64/fabricmanager-linux-x86_64-{{ .NVIDIA_DRIVER_LTS_VERSION }}-archive.tar.xz
         destination: fabricmanager.tar.xz
         sha256: {{ .NVIDIA_FABRIC_MANAGER_LTS_AMD64_SHA256 }}
         sha512: {{ .NVIDIA_FABRIC_MANAGER_LTS_AMD64_SHA512 }}
-    # {{ end }} This in fact is YAML comment, but Go templating instruction is evaluated by bldr
+      - url: https://developer.download.nvidia.com/compute/cuda/redist/nvlsm/linux-x86_64/nvlsm-linux-x86_64-{{ .NVIDIA_NVLSM_VERSION }}-archive.tar.xz
+        destination: nvlsm.tar.xz
+        sha256: {{ .NVIDIA_NVLSM_AMD64_SHA256 }}
+        sha512: {{ .NVIDIA_NVLSM_AMD64_SHA512 }}
+    # {{ end }} This in fact is YAML comment, but Go templating instruction is evaluated by bld
     prepare:
       - |
-        tar -xf fabricmanager.tar.xz --strip-components=1
-
+        mkdir fm sm
+        tar -xf fabricmanager.tar.xz --strip-components=1 -C fm
+        tar -xf nvlsm.tar.xz --strip-components=1 -C sm
+      - |
         sed -i 's#$VERSION#{{ .VERSION }}#' /pkg/manifest.yaml
     install:
       - |
-        mkdir -p /rootfs/usr/local/bin \
-          /rootfs/usr/local/lib \
-          /rootfs/usr/local/share/nvidia/nvswitch \
-          /rootfs/usr/local/lib/containers/nvidia-fabricmanager \
-          /rootfs/usr/local/etc/containers
+        mkdir -p /rootfs/usr/local/lib/containers/nvidia-fabricmanager/usr/bin \
+                 /rootfs/usr/local/lib/containers/nvidia-fabricmanager/usr/lib \
+                 /rootfs/usr/local/lib/containers/nvidia-fabricmanager/usr/share/nvidia/nvswitch \
+                 /rootfs/usr/local/lib/containers/nvidia-fabricmanager/opt/nvidia/nvlsm/sbin \
+                 /rootfs/usr/local/lib/containers/nvidia-fabricmanager/opt/nvidia/nvlsm/lib \
+                 /rootfs/usr/local/lib/containers/nvidia-fabricmanager/usr/share/nvidia/nvlsm
+      # nvlsm
+      - |
+        cp sm/sbin/nvlsm /rootfs/usr/local/lib/containers/nvidia-fabricmanager/opt/nvidia/nvlsm/sbin/
 
-        cp lib/libnvfm.so.1 /rootfs/usr/local/lib/libnvfm.so.1
-        ln -s libnvfm.so.1 /rootfs/usr/local/lib/libnvfm.so
+        cp sm/lib/libgrpc_mgr.so \
+           /usr/lib/libgcc_s.so.1 \
```

Comment:
why are we copying `libgcc`? is it to contain fabric manager to itself? If copying libgcc we might need to copy other so files too

Reply:
Also unless I'm mistaken, `libgcc_s.so` is nowhere to be found in Talos. It's not part of `misc/glibc`. So the move to self-contain fabricmanager is not involved in this decision. It's purely because newly introduced NVLSM needs it

Reply:
I wonder what brings in `libgcc_s.so`

Reply:
> And that's why I introduced Wolfi here, because it's versionned and make for a stable build

this is not true, while wolfi image is pinned, any `apk add` would bring in a new version of package.

Reply:
> We really need to test this in CI with the DGX/HGX machines

Agreed. All I can tell you is it runs fine in production for Blackwell DGX with the current state of this PR applied to 1.10.4

Reply:
> this is not true, while wolfi image is pinned, any `apk add` would bring in a new version of package.

Oh.. of course I can see why now. I'll dig deeper into the build system to find a solution